internal/pkg/pluginengine/helper.go

```diff
@@ -6,19 +6,12 @@ import (
 	"plugin"
 	"time"
 
-	"github.com/spf13/viper"
-
 	"github.com/merico-dev/stream/internal/pkg/configloader"
 	"github.com/merico-dev/stream/internal/pkg/planmanager"
 	"github.com/merico-dev/stream/internal/pkg/statemanager"
 )
 
-func loadPlugin(tool *configloader.Tool) (DevStreamPlugin, error) {
-	pluginDir := viper.GetString("plugin-dir")
-	if pluginDir == "" {
-		return nil, fmt.Errorf("plugin-dir is \"\"")
-	}
-
+func loadPlugin(pluginDir string, tool *configloader.Tool) (DevStreamPlugin, error) {
```

Comment:
I added the pluginDir param here, because `IsHealthy()` is called by `smoke`, and the `smoke` has no --plugin-dir flag to use. So `IsHealthy()` needs to call `loadPlugin()` with "pluginDir" info.